example/grammar/magnet.cpp

```diff
@@ -0,0 +1,496 @@
+//
+// Copyright (c) 2022 alandefreitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+//
+
+#include <boost/url/url_view.hpp>
+#include <boost/url/rfc/absolute_uri_rule.hpp>
+#include <boost/url/grammar/charset.hpp>
+#include <iostream>
+
+using namespace boost::urls;
+
+// A small filter and transform view
+template <class C, class V, class P, class T>
+class filter_view
```

Comment:
Hmm I think this needs to be broken up into individual .hpp files

Reply:
Can we do that? I thought examples were supposed to be in a single file.

Reply:
As an author you can exercise editorial discretion on whether or not to break things up. A simple HTTP client or server is nice to have as one file that you can just copy. But you're creating a robust vocabulary component with additional utility classes that could be useful on their own (e.g. `filter_view`). In this case I think multiple files are warranted. But its up to you.

Reply:
Mmm... OK. I saw Boost.Beast examples are huge. So I thought it was kind of a rule not to split examples into multiple files. No problem then.